src/EnergyPlus/DaylightingManager.cc

```diff
@@ -1326,266 +1265,186 @@ void CalcDayltgCoeffsMapPoints(EnergyPlusData &state, int const ZoneNum)
-                                                                         PHRAY,
-                                                                         LSHCAL,
-                                                                         InShelfSurf,
-                                                                         COSB,
-                                                                         ObTrans,
-                                                                         TVISB,
-                                                                         DOMEGA,
-                                                                         ICtrl,
-                                                                         ShType,
-                                                                         BlNum,
-                                                                         THRAY,
-                                                                         WNORM2,
-                                                                         ExtWinType,
-                                                                         IConst,
-                                                                         AZVIEW,
-                                                                         RREF2,
-                                                                         hitIntObs,
-                                                                         hitExtObs,
-                                                                         DataDaylighting::iCalledFor::MapPoint,
-                                                                         TVISIntWin,
-                                                                         TVISIntWinDisk,
-                                                                         MapNum,
-                                                                         MapWindowSolidAngAtRefPtWtd);
-                            } // End of hourly sun position loop, IHR
-                        } else {
-                            if (state.dataEnvrn->SunIsUp && !MySunIsUpFlag) {
-                                ISunPos = 0;
-                                MySunIsUpFlag = true;
-                            } else if (state.dataEnvrn->SunIsUp && MySunIsUpFlag) {
-                                ISunPos = 1;
-                            } else if (!state.dataEnvrn->SunIsUp && MySunIsUpFlag) {
-                                MySunIsUpFlag = false;
-                                ISunPos = -1;
-                            } else if (!state.dataEnvrn->SunIsUp && !MySunIsUpFlag) {
-                                ISunPos = -1;
-                            }
+        for (loopwin = 1; loopwin <= thisEnclDaylight.NumOfDayltgExtWins; ++loopwin) {
+
+            // daylightingCtrlNum parameter is unused for map points
+            FigureDayltgCoeffsAtPointsSetupForWindow(state,
```

Comment:
This function has 20+ parameters. Not a cheap function to call when x86_64 has only 6 parameter passing registers.  Probably an item for another day. See [here](https://github.com/NREL/EnergyPlus/wiki/Parameters-and-Return-Values) for ideas.